### ISSUE TITLE
Reduce watch update spam from rancher local-path-provisioner

### ISSUE THREAD

**Issue by jayunit100** [kubernetes-sigs/kind#2453]
# Workaround

quick note if you dont want spammy EndpointSliceUpdate logs, just delete the `local-path-provisioner` but obviously that would break your pvcs in a vanilla kind cluster

# The issue
On any kube-proxy running in kind, the local-path-provisioner for rancher makes a bunch of noise
```
I0907 13:28:05.246094       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:28:24.219829       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:28:24.261791       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:28:41.276373       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:28:41.297670       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:41:41.575654       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:41:41.575863       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:56:41.098421       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 13:56:41.098677       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 14:11:40.401634       1 config.go:260] Calling handler.OnEndpointSliceUpdate
I0907 14:11:40.401676       1 config.go:260] Calling handler.OnEndpointSliceUpdate
```
It turns out the reason is that its constantly updating the leave value, as we can see if we rapidly capture the endpoints and diff them, we see they are *constantly* changing for a trivial reason... 
```
➜  ~ kubectl get endpoints -A -o yaml > /tmp/11
➜  ~ kubectl get endpoints -A -o yaml > /tmp/22
➜  ~ diff /tmp/11 /tmp/22
73c73
<       control-plane.alpha.kubernetes.io/leader: '{"holderIdentity":"local-path-provisioner-547f784dff-kmxpw_0fbc4e31-f0fd-4ab4-b578-ea80baf4295c","leaseDurationSeconds":15,"acquireTime":"2021-09-03T21:20:26Z","renewTime":"2021-09-07T14:19:01Z","leaderTransitions":0}'
---
>       control-plane.alpha.kubernetes.io/leader: '{"holderIdentity":"local-path-provisioner-547f784dff-kmxpw_0fbc4e31-f0fd-4ab4-b578-ea80baf4295c","leaseDurationSeconds":15,"acquireTime":"2021-09-03T21:20:26Z","renewTime":"2021-09-07T14:19:07Z","leaderTransitions":0}'
77c77
<     resourceVersion: "612915"
---
>     resourceVersion: "612929"
```

**What should be cleaned up or changed**:

It will make the logs for the kube-proxy easier to read and less confusing / take up less space if we can get rid of the continuous calls to `OnEndpointsUpdate`. 

**Why is this needed**:

See above reason.

Probably a good-first-issue ? I bet the solution would be to set rancher's installation to have a node-update-lease of like 120 seconds or something (or maybe longer, after all this is `kind` and we don't expect people to need realtime HA storage recovery on it).

**Comment by liggitt** [comment]
local-path-provisioner should transition to using Lease objects for leases instead of endpoints

**Comment by aojea** [comment]
great catch, this is coming from

https://github.com/kubernetes-sigs/kind/blob/f8e6aa668edd337fa917e7ed9e0eab1deeb1d7af/pkg/build/nodeimage/const_storage.go#L19-L30

**Comment by aojea** [comment]
since we run a singleton we can just disable leader election at all :)
https://github.com/kubernetes-sigs/kind/blob/f8e6aa668edd337fa917e7ed9e0eab1deeb1d7af/pkg/build/nodeimage/const_storage.go#L80

**Comment by liggitt** [comment]
separately, an issue should be spawned for *that* component to transition to let the lock resource be configurable so consumers can safely transition from endpoints → endpointsleases → leases

https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/master/controller/controller.go#L867-L878

**Comment by liggitt** [comment]
>   replicas: 1 

replica of 1 doesn't guarantee there will never be concurrent instances... especially in cases where a new revision of the deployment is rolling out

**Comment by snOm3ad** [comment]
Hello! I'd like to get assigned this issue if no one else has taken it. This would be my first time contributing to this project!

**Comment by aojea** [comment]
This is a long road as Jordan explained in previous comments, we should start here 
https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/issues/111

Once that is fixed I'm happy to guide you

**Comment by BenTheElder** [comment]
This seems to still be blocked on https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/issues/111

Maybe we can work around this in the meantime?
We should also probably cross-track in the rancher repo.

**Comment by aojea** [comment]
@NikhilSharmaWe https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/issues/111#issuecomment-990751321

**Comment by BenTheElder** [comment]
this is going to be solved indirectly by the latest version simply disabling leader election entirely as suggested in https://github.com/kubernetes-sigs/kind/issues/2453#issuecomment-914391519, upstream local-path-provisioner has actually opted to do this

updating to the latest in https://github.com/kubernetes-sigs/kind/pull/2762

**Comment by BenTheElder** [comment]
should be fixed in v0.14.0 (just released)
